.gitlab-ci.yml

```diff
@@ -183,3 +185,25 @@ parthenon-cpu-performance_and_regression-mpi:
       - build-perf-mpi/CMakeFiles/CMakeOutput.log
       - build-perf-mpi/tst/regression/outputs/advection_convergence/advection-errors.dat
       - build-perf-mpi/tst/regression/outputs/advection_convergence/advection-errors.png
+
+# run unit suite on CPUs with code coverage
+parthenon-cpu-coverage:
+  tags:
+    - cpu
+  stage: coverage
+  script:
+    - mkdir build-debug-coverage
+    - cd build-debug-coverage
+    - cmake -DCMAKE_BUILD_TYPE=Debug -DHDF5_ROOT=/usr/local/hdf5/parallel
+      -DKokkos_ENABLE_OPENMP=True -DKokkos_ARCH_WSM=True
+      -DMPIEXEC_PREFLAGS="--allow-run-as-root" -DCODE_COVERAGE=ON
+      ../ && make -j${J} && make coverage && make coverage-upload
+  artifacts:
+    when: always
+    expire_in: 3 days
+    paths:
+      - build-debug-coverage/CMakeFiles/CMakeOutput.log
+  only:
+    - master
```

Comment:
Is there any way to get code coverage for every MR? Too expensive?

Reply:
Sure.

Reply:
Are you meaning if not merging to master but some other branch?